### DISCUSSION TITLE
Advice for PR to Add Selenium With Docker

### DISCUSSION THREAD

**Discussion by BenA-SA** [cookiecutter/cookiecutter-django#4397] [Ideas]
Hi,

I have used cookiecutter-django several times and it's a fantastic project, so thanks very much to the people who have created it.

I have managed to add Selenium within a dockerized cookiecutter-django and used it multiple times, and was wondering if there would be any interest in me making a PR to include it? Or is it not included already for a reason?

If this is something that could be added, what steps should I be aware of in trying to make a successful PR?

Thanks!

**Comment by browniebroke** [comment]
Is that a big change, in that, does it touch many files or is it mainly the case of adding a new service to the compose file and a new job in CI? If you have an example of a PR on a generated project, that would probably help with answering that question. My main concern is how easy it would be to maintain over time. Things we have to factor in:

- Is there anything bound to a version of Django or Python?
- Would it be easy to remove? Easy to add?
- Can it run on any CI or would it be a partial support?
- Is it commonly used enough? While Selenium has its use cases, I don't use it a lot. In my experience (from a few years back) it's quite a heavy dependency and it can be flaky over time. 
- Are there any competing alternative that people might prefer? I've recently came across lighter alternatives to run e2e tests (Playwright, Cypress) which might be requested by folks if we start to add Selenium support, and supporting multiple framework might be too tricky.

Also, when adding this on a generated project is fairly straightforward, it's sometimes good to keep it out, and let the maintenance focus our efforts on more core integrations. We're a limited team of volunteers.
